client/src/Connection.ts

```diff
@@ -72,10 +72,20 @@ export default class Connection {
   }
 }
 
+interface CallbackCommands {
+  [command: string]: (data: any) => void;
+}
+
+type Callbacks = CallbackCommands & {
+  open?: (evt: Event) => void;
+  close?: (evt: Event) => void;
+  socketError?: (evt: Event) => void;
+};
+
 const separator = ' ';
 const unpack = (data: string) => {
   const name = data.split(separator)[0];
   return [name, data.substring(name.length + 1, data.length)];
 };
-const unmarshal = (data: string) => JSON.parse(data);
+const unmarshal = (data: string) => JSON.parse(data) as unknown;
```

Comment:
"unknown" is a much more restrictive version of any that only works on the right side of an assignment
https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type